src/tests/unit/fixtures/thrift-server/annotations_struct.solution.ts

```diff
@@ -0,0 +1,112 @@
+export interface IMyStruct {
+    id: number;
+    bigID: thrift.Int64;
+}
+export interface IMyStructArgs {
+    id?: number;
+    bigID?: number | thrift.Int64;
+}
+export const MyStructCodec: thrift.IStructCodec<IMyStructArgs, IMyStruct> = {
+    encode(args: IMyStructArgs, output: thrift.TProtocol): void {
+        const obj = {
+            id: (args.id != null ? args.id : 45),
+            bigID: (args.bigID != null ? (typeof args.bigID === "number" ? new thrift.Int64(args.bigID) : args.bigID) : thrift.Int64.fromDecimalString("23948234"))
+        };
+        output.writeStructBegin("MyStruct");
+        if (obj.id != null) {
+            output.writeFieldBegin("id", thrift.TType.I32, 1);
+            output.writeI32(obj.id);
+            output.writeFieldEnd();
+        }
+        if (obj.bigID != null) {
+            output.writeFieldBegin("bigID", thrift.TType.I64, 2);
+            output.writeI64(obj.bigID);
+            output.writeFieldEnd();
+        }
+        output.writeFieldStop();
+        output.writeStructEnd();
+        return;
+    },
+    decode(input: thrift.TProtocol): IMyStruct {
+        let _args: any = {};
+        input.readStructBegin();
+        while (true) {
+            const ret: thrift.IThriftField = input.readFieldBegin();
+            const fieldType: thrift.TType = ret.fieldType;
+            const fieldId: number = ret.fieldId;
+            if (fieldType === thrift.TType.STOP) {
+                break;
+            }
+            switch (fieldId) {
+                case 1:
+                    if (fieldType === thrift.TType.I32) {
+                        const value_1: number = input.readI32();
+                        _args.id = value_1;
+                    }
+                    else {
+                        input.skip(fieldType);
+                    }
+                    break;
+                case 2:
+                    if (fieldType === thrift.TType.I64) {
+                        const value_2: thrift.Int64 = input.readI64();
+                        _args.bigID = value_2;
+                    }
+                    else {
+                        input.skip(fieldType);
+                    }
+                    break;
+                default: {
+                    input.skip(fieldType);
+                }
+            }
+            input.readFieldEnd();
+        }
+        input.readStructEnd();
+        if (_args.id !== undefined && _args.bigID !== undefined) {
+            return {
+                id: (_args.id != null ? _args.id : 45),
+                bigID: (_args.bigID != null ? _args.bigID : thrift.Int64.fromDecimalString("23948234"))
+            };
+        }
+        else {
+            throw new thrift.TProtocolException(thrift.TProtocolExceptionType.UNKNOWN, "Unable to read MyStruct from input");
+        }
+    }
+};
+export class MyStruct extends thrift.StructLike implements IMyStruct {
+    public id: number = 45;
```

Comment:
why did we assign default values to MyStruct?

Reply:
These are the user-defined defaults. We set them here and override them in the constructor if they pass in other values.